src/mokernels/moinput.jl

```diff
@@ -1,15 +1,44 @@
 """
-    MOInput(x::AbstractVector, out_dim::Integer)
+    IsotopicByFeatures(x::AbstractVector, out_dim::Integer)
 
-A data type to accomodate modelling multi-dimensional output data.
+`IsotopicByFeatures(x, out_dim)` has length `out_dim * length(x)`.
 
-`MOInput(x, out_dim)` has length `length(x) * out_dim`.
+```jldoctest
+julia> x = [1, 2, 3];
+
+julia> IsotopicByFeatures(x, 2)
+6-element IsotopicByFeatures{Int64, Vector{Int64}}:
+ (1, 1)
+ (1, 2)
+ (2, 1)
+ (2, 2)
+ (3, 1)
+ (3, 2)
+```
+
+Accommodates modelling multi-dimensional output data where all outputs are always observed.
+
+As shown above, an `IsotopicByFeatures` represents a vector of tuples.
+The first `out_dim` elements represent all outputs for the first input, the second
+`out_dim` elements represent the outputs for the second input, etc.
+
+See [Inputs for Multiple Outputs](@ref) in the docs for more info.
+"""
+struct IsotopicByFeatures{S,T<:AbstractVector{S}} <: AbstractVector{Tuple{S,Int}}
+    x::T
+    out_dim::Int
```

Comment:
```suggestion
    output_dim::Int
```
?
(both for the other struct as well of course)

Reply:
@thomasgudjonwright I would be happy for this change to be made in this PR, since this is a new type. Are you happy to make it?

Reply:
Resolving along with thread above for reason discussed in above thread. Discussion noted in a new [issue](https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/issues/332)

Reply:
@willtebbutt do we want any disparity between the two types though? Since we couldn't do the same for `IsotopicByOutputs` without breaking any `MOInputs` currently in the wild, right?